internal/azure/exchanger.go

```diff
@@ -0,0 +1,81 @@
+package azure
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type tokenResponse struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	Resource     string `json:"resource"`
+	TokenType    string `json:"token_type"`
+}
+
+type Exchanger struct {
+	scheme   string
```

Comment:
Doesn't seem like something that would ever be changed or overridden, should be safe to hardcode this as `https` in the resulting URL.